[CI] Zen1IT.testMixedClusterFormation failure

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+internalClusterTest/2440/console

I cannot reproduce this on 7.0 currently, but here's the details
```
./gradlew :server:integTest \
  -Dtests.seed=196AC143B1CD91F6 \
  -Dtests.class=org.elasticsearch.cluster.coordination.Zen1IT \
  -Dtests.method="testMixedClusterFormation" \
  -Dtests.security.manager=true \
  -Dtests.locale=sv-SE \
  -Dtests.timezone=America/Rainy_River \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

```
java.lang.AssertionError: cluster state size does not match expected:<1882> but was:<1836>
	at __randomizedtesting.SeedInfo.seed([196AC143B1CD91F6:CEE18F2BCE07D48]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.elasticsearch.test.ESIntegTestCase.ensureClusterStateConsistency(ESIntegTestCase.java:1161)
	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:577)
	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2196)
```

--- a/server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java
+++ b/server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java
@@ -465,6 +465,12 @@ public class ClusterApplierService extends AbstractLifecycleComponent implements
 
         nodeConnectionsService.disconnectFromNodesExcept(newClusterState.nodes());
 
+        assert newClusterState.coordinationMetaData().getLastAcceptedConfiguration()
+            .equals(newClusterState.coordinationMetaData().getLastCommittedConfiguration())
+            : newClusterState.coordinationMetaData().getLastAcceptedConfiguration()
+            + " vs " + newClusterState.coordinationMetaData().getLastCommittedConfiguration()
+            + " on " + newClusterState.nodes().getLocalNode();
+
         logger.debug("set locally applied cluster state to version {}", newClusterState.version());
         state.set(newClusterState);
 

--- a/server/src/main/java/org/elasticsearch/discovery/zen/PendingClusterStatesQueue.java
+++ b/server/src/main/java/org/elasticsearch/discovery/zen/PendingClusterStatesQueue.java
@@ -22,6 +22,8 @@ package org.elasticsearch.discovery.zen;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.coordination.CoordinationMetaData;
+import org.elasticsearch.cluster.metadata.MetaData;
 import org.elasticsearch.cluster.node.DiscoveryNode;
 
 import java.util.ArrayList;
@@ -246,7 +248,16 @@ public class PendingClusterStatesQueue {
             }
         }
         assert stateToProcess.committed() : "should only return committed cluster state. found " + stateToProcess.state;
-        return stateToProcess.state;
+
+        final ClusterState committedState = stateToProcess.state;
+        final CoordinationMetaData coordinationMetaData = committedState.coordinationMetaData();
+        if (coordinationMetaData.getLastAcceptedConfiguration().equals(coordinationMetaData.getLastCommittedConfiguration())) {
+            return committedState;
+        } else {
+            return ClusterState.builder(committedState).metaData(MetaData.builder(committedState.metaData())
+                .coordinationMetaData(CoordinationMetaData.builder(coordinationMetaData)
+                    .lastCommittedConfiguration(committedState.getLastAcceptedConfiguration()).build())).build();
+        }
     }
 
     /** returns all pending states, committed or not */

